File: wrappers/dotnet/indy-sdk-dotnet-test/IndyIntegrationTestWithSingleWallet.cs
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test
{
    public abstract class IndyIntegrationTestWithSingleWallet : IndyIntegrationTestBase
    {
        protected Wallet wallet;

        [TestInitialize]
        public async Task CreateWallet()
        {
            await Wallet.CreateWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
            wallet = await Wallet.OpenWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
        }

        [TestCleanup]
        public async Task DeleteWallet()
        {
            await wallet.CloseAsync();
            await Wallet.DeleteWalletAsync(WALLET_CONFIG, WALLET_CREDENTIALS);
        }
    }

}

File: wrappers/android/src/androidTest/java/org/hyperledger/indy/sdk/IndyIntegrationTestWithSingleWallet.java
package org.hyperledger.indy.sdk;

import org.hyperledger.indy.sdk.wallet.Wallet;
import org.junit.After;
import org.junit.Before;


public class IndyIntegrationTestWithSingleWallet extends IndyIntegrationTest {

	public Wallet wallet;

	@Before
	public void createWallet() throws Exception {
		Wallet.createWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
		this.wallet = Wallet.openWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}

	@After
	public void deleteWallet() throws Exception {
		wallet.closeWallet().get();
		Wallet.deleteWallet(WALLET_CONFIG, WALLET_CREDENTIALS).get();
	}
}
